vault/resource_mongodbatlas_secret_role.go

```diff
@@ -0,0 +1,238 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package vault
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-vault/internal/consts"
+	"github.com/hashicorp/terraform-provider-vault/internal/provider"
+)
+
+func mongodbAtlasSecretRoleResource() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: mongodbAtlasSecretRoleCreateUpdate,
+		ReadContext:   ReadContextWrapper(mongodbAtlasSecretRoleRead),
+		UpdateContext: mongodbAtlasSecretRoleCreateUpdate,
+		DeleteContext: mongodbAtlasSecretRoleDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			consts.FieldBackend: {
+				Type:         schema.TypeString,
+				Required:     true,
+				Description:  "Path where MongoDB Atlas secret backend is mounted",
+				ValidateFunc: provider.ValidateNoLeadingTrailingSlashes,
+			},
+			consts.FieldPath: {
+				Type:         schema.TypeString,
+				Required:     true,
+				Description:  "Path where MongoDB Atlas backend is mounted",
+				ValidateFunc: provider.ValidateNoLeadingTrailingSlashes,
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Name of the role",
+			},
+			"organization_id": {
+				Type:        schema.TypeString,
+				Required:    false,
```

Comment:
These are probably failing in CI build right now because currently the fields are neither Required, Optional or Computed. Those errors should go away when we update these to be `Optional: true` instead of `Required: false`

Reply:
Makes sense, fixed now. Thank you!